internal/hook/item_hook_handler.go

```diff
@@ -205,23 +205,24 @@ func getPodExecHookFromAnnotations(annotations map[string]string, phase hookPhas
 	}
 }
 
-type resourceHook struct {
-	name          string
-	namespaces    *collections.IncludesExcludes
-	resources     *collections.IncludesExcludes
-	labelSelector labels.Selector
-	pre           []api.BackupResourceHook
-	post          []api.BackupResourceHook
+// ResourceHook is a hook for a given resource.
+type ResourceHook struct {
```

Comment:
suggest qualifying as `BackupResourceHook`

Reply:
This specifically I left generic because I was thinking it would be re-usable for the non-initContainer restore hooks.

I could keep this one and split out a Backup one that includes the Pre/Post fields.

Reply:
Yeah. We can break this up and do the refactor work as part of restore hook implementation.
Keeping it the way it is now wfm.

Reply:
👍 I'm cool with that plan - keeping it for now, but doing some clarification work when it comes time to actually add new types.